contxt/cli/commands/iot.py

```diff
@@ -113,3 +115,193 @@ def data(
     print(f"Writing data to {output}...")
     flat_data = [{"timestamp": k, **v} for k, v in data.items()]
     Serializer.to_csv(flat_data, output)
+
+
+@iot.command()
+@click.argument("feed_key", type=str)
+@click.argument("worksheet_file", type=click.File("r"))
+@click.pass_obj
+def ingest_worksheet(clients: Clients, feed_key, worksheet_file) -> None:
+    """Field Worksheet Ingestor"""
+    feed = clients.iot.get_feed_with_key(feed_key)
+    if not feed:
+        print(f"Feed with key {feed_key} does not exist")
+        return
+    groupings_by_label = {}
+    groupings = clients.iot.get_field_groupings_for_facility(feed.facility_id)
+    for grouping in groupings:
+        groupings_by_label[grouping.label] = grouping
+
+    field_objs = []
+    fields_for_feed = clients.iot.get_fields_for_feed(feed_id=feed.id)
+    fields_descriptors_for_feed = [f.field_descriptor for f in fields_for_feed]
+
+    desired_groupings_by_field = {}
+    with worksheet_file as f:
+
+        headers = [
+            "Field Descriptor",
+            "Label",
+            "Data Type",
+            "Units",
+            "Scalar",
+            "Cumulative Value?",
+            "equipment group",
+            "Facility Main?",
+        ]
+        reader = csv.DictReader(f, fieldnames=headers)
+
+        # skip the header
+        next(reader)
+
+        # create a bunch of field objects
+        for row in reader:
+            if row["Label"].replace(" ", "").startswith("-"):
+                label = row["equipment group"] + row["Label"]
+            else:
+                label = row["Label"]
+
+            if row["Field Descriptor"] in fields_descriptors_for_feed:
+                print(f'Already provisioned field with descriptor {row["Field Descriptor"]}')
+            else:
+                field_objs.append(
+                    Field(
+                        label=label,
+                        output_id=feed.id,
+                        field_descriptor=row["Field Descriptor"],
+                        units=row["Units"],
+                        field_human_name=row["Field Descriptor"],
+                        feed_key=feed_key,
+                        value_type=FieldValueType("numeric"),
```

Comment:
why hardcode this?

Reply:
I'm not sure, and John hasn't had the time to talk to me about these changes. This was another bit that I simply replicated from the original PR.

@jrcrawfo Maybe you could answer why you originally decided to hardcode this.

Reply:
fixed this